aws/structure.go

```diff
@@ -3035,6 +3035,23 @@ func flattenCognitoIdentityPoolRoles(config map[string]*string) map[string]strin
 	return m
 }
 
+func expandCognitoIdentityProviderMap(config map[string]interface{}) map[string]*string {
+	m := map[string]*string{}
+	for k, v := range config {
+		s := v.(string)
+		m[k] = &s
+	}
+	return m
+}
+
+func flattenCognitoIdentityProviderMap(config map[string]*string) map[string]string {
```

Comment:
The AWS SDK already provides a helper function to perform this: `aws.StringValueMap()`